cli/azd/pkg/azdo/service_connection.go

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azdo
+
+import (
+	"context"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/microsoft/azure-devops-go-api/azuredevops"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/build"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/serviceendpoint"
+)
+
+// authorize a service connection to be used in all pipelines
+func authorizeServiceConnectionToAllPipelines(
+	ctx context.Context,
+	projectId string,
+	endpoint *serviceendpoint.ServiceEndpoint,
+	connection *azuredevops.Connection) error {
+	buildClient, err := build.NewClient(ctx, connection)
+	if err != nil {
+		return err
+	}
+
+	endpointResource := "endpoint"
+	endpointAuthorized := true
+	endpointId := endpoint.Id.String()
+	resources := make([]build.DefinitionResourceReference, 1)
+	resources[0] = build.DefinitionResourceReference{
+		Type:       &endpointResource,
+		Authorized: &endpointAuthorized,
+		Id:         &endpointId,
+	}
```

Comment:
Nit, you can just define this inline without have to make the slice with fixed length.

ex)
```go
resources := []build.DefinitionResourceReference{
    build.DefinitionResourceReference{
		Type:       &endpointResource,
		Authorized: &endpointAuthorized,
		Id:         &endpointId,
	}
}
```

Reply:
done